### PR TITLE
Rebuild for CUDA 12

### PR DESCRIPTION
**Pull request by RaulPPelaez**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by RaulPPelaez** [comment]
@conda-forge-admin, please rerender

**Comment by RaulPPelaez** [comment]
I updated the meta.yaml to install the cuda libraries in the new way. I also added myself as a maintainer.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -37,6 +43,16 @@ requirements:
...
+    {% if cuda_major >= 12 %}
+    - cuda-driver-dev
+    - cuda-cudart-dev
+    - cuda-nvrtc-dev
+    - cuda-nvtx-dev
+    - cuda-nvml-dev
+    - libcublas-dev
+    - libcufft-dev
+    - libcurand-dev
+    {% endif %}
```
Comment (peastman) [inline recipe/meta.yaml:55]: The only ones of these we use are driver, nvrtc, and cufft.

Comment (mikemhenry) [reply recipe/meta.yaml]: I just pushed https://github.com/conda-forge/openmm-torch-feedstock/pull/43/commits/2d888d1740cc6a9957f809b7bc9fc90822781b72 to see if we can trim the deps down

**Comment by mikemhenry** [comment]
Looks like trimming things down worked

**Comment by mikemhenry** [comment]
Going to merge this so we can keep the migration going

**Pull request merged**